model/task/task.go

```diff
@@ -909,20 +907,23 @@ func (t *Task) cacheExpectedDuration() error {
 // to a pod.
 func (t *Task) MarkAsContainerDispatched(ctx context.Context, env evergreen.Environment, podID, agentVersion string) error {
 	dispatchedAt := time.Now()
+	output := t.newTaskOutput()
+
 	query := ScheduledContainerTasksQuery()
 	query[IdKey] = t.Id
 	query[StatusKey] = evergreen.TaskUndispatched
 	query[ContainerAllocatedKey] = true
-	update := bson.M{
-		"$set": bson.M{
-			StatusKey:        evergreen.TaskDispatched,
-			DispatchTimeKey:  dispatchedAt,
-			LastHeartbeatKey: dispatchedAt,
-			PodIDKey:         podID,
-			AgentVersionKey:  agentVersion,
-		},
+	set := bson.M{
+		StatusKey:        evergreen.TaskDispatched,
+		DispatchTimeKey:  dispatchedAt,
+		LastHeartbeatKey: dispatchedAt,
+		PodIDKey:         podID,
+		AgentVersionKey:  agentVersion,
+	}
+	if output != nil {
```

Comment:
I did this to be safe since `newTaskOutput()` _can_ return `nil`, and who knows if in the future there will be another case where it doesn't make sense to have a task output for a task document.